### PR TITLE
Restore old Slack engine, add dreprecation, rename new Slack engine

### PR DESCRIPTION
**Pull request by garethgreenaway**
### What does this PR do?
Restore the previous slack engine and deprecate it, rename replace the slack engine to slack_bolt until deprecation

### What issues does this PR fix or reference?
Fixes:  #62957

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [x] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: changelog/63095.added [hunk 1]
```diff
@@ -0,0 +1 @@
+Restore the previous slack engine and deprecate it, rename replace the slack engine to slack_bolt until deprecation
```
Comment (Ch3LL) [inline changelog/63095.added:1]: We should probably add a separate `.deprecated` changelog for noting the slack engine being depprecated as well as the added

File: salt/engines/slack_bolt_engine.py [hunk 2]
```diff
@@ -0,0 +1,1004 @@
+"""
+An engine that reads messages from Slack and can act on them
+
+.. versionadded:: 2016.3.0
```
Comment (Ch3LL) [inline salt/engines/slack_bolt_engine.py:4]:
```suggestion
.. versionadded:: 3006.0
```

**Review by Ch3LL — CHANGES_REQUESTED** [review]

**Comment by Ch3LL** [comment]
I'm also not seeing slack_bolt in our tests requirements so I'm guesing the tests are getting skipped

**Comment by garethgreenaway** [comment]
@Ch3LL The tests will only run if the two Slack Python libraries are installed, the newer one we can install but the older one has difficulties with newer versions of Python.

File: salt/engines/slack_bolt_engine.py [hunk 3]
```diff
@@ -0,0 +1,1004 @@
...
+                )
+            )  # Boom!
+        while self._run_until():
+            while self.msg_queue:
+                msg = self.msg_queue.popleft()
+                try:
+                    msg_text = self.message_text(msg)
+                except (ValueError, TypeError) as msg_err:
+                    log.debug(
+                        "Got an error from trying to get the message text %s", msg_err
```
Comment (twangboy) [inline salt/engines/slack_bolt_engine.py:627]: Got an error trying

File: salt/engines/slack_bolt_engine.py [hunk 4]
```diff
@@ -0,0 +1,1004 @@
...
+                    continue
+            yield {"done": True}
+
+    def get_target(self, permitted_group, cmdline, alias_cmdline):
+        """
+        When we are permitted to run a command on a target, look to see
+        what the default targeting is for that group, and for that specific
+        command (if provided).
+
+        It's possible for None or False to be the result of either, which means
```
Comment (twangboy) [inline salt/engines/slack_bolt_engine.py:682]:
```
``None`` or ``False``
```

File: salt/engines/slack_bolt_engine.py [hunk 5]
```diff
@@ -0,0 +1,1004 @@
...
+        that it's expected that the caller provide a specific target.
+
+        If no configured target is provided, the command line will be parsed
+        for target=foo and tgt_type=bar
+
```
Comment (twangboy) [inline salt/engines/slack_bolt_engine.py:687]: args need to be documented

File: salt/engines/slack_bolt_engine.py [hunk 6]
```diff
@@ -0,0 +1,1004 @@
...
+        Run each of them through ``get_configured_target(('foo', f), 'pillar.get')`` and confirm a valid target
+
+        """
+        # Default to targeting all minions with a type of glob
+        null_target = {"target": "*", "tgt_type": "glob"}
+
+        def check_cmd_against_group(cmd):
+            """
+            Validate cmd against the group to return the target, or a null target
```
Comment (twangboy) [inline salt/engines/slack_bolt_engine.py:709]: Document `cmd`

File: salt/engines/slack_bolt_engine.py [hunk 7]
```diff
@@ -0,0 +1,1004 @@
...
+            log.debug("this cmdline has target %s.", this_cl)
+            if checked.get("target"):
+                return checked
+        return null_target
+
+    def format_return_text(
+        self, data, function, **kwargs
+    ):  # pylint: disable=unused-argument
+        """
+        Print out YAML using the block mode
```
Comment (twangboy) [inline salt/engines/slack_bolt_engine.py:744]: Docs for data and function

File: salt/engines/slack_bolt_engine.py [hunk 8]
```diff
@@ -0,0 +1,1004 @@
...
+
+            log.exception(
+                "Exception encountered when trying to serialize %s",
+                pprint.pformat(data),
+            )
+            return "Got an error trying to serialze/clean up the response"
+
+    def parse_args_and_kwargs(self, cmdline):
+        """
+        cmdline: list
```
Comment (twangboy) [inline salt/engines/slack_bolt_engine.py:769]: Maybe a description as well for cmdline

File: salt/engines/slack_bolt_engine.py [hunk 9]
```diff
@@ -0,0 +1,1004 @@
...
+        return (args, kwargs)
+
+    def get_jobs_from_runner(self, outstanding_jids):
+        """
+        Given a list of job_ids, return a dictionary of those job_ids that have
+        completed and their results.
+
+        Query the salt event bus via the jobs runner. jobs.list_job will show
+        a job in progress, jobs.lookup_jid will return a job that has
+        completed.
```
Comment (twangboy) [inline salt/engines/slack_bolt_engine.py:793]: Docs for the arg

File: salt/engines/slack_bolt_engine.py [hunk 10]
```diff
@@ -0,0 +1,1004 @@
...
+
+        return results
+
+    def run_commands_from_slack_async(
+        self, message_generator, fire_all, tag, control, interval=1
+    ):
+        """
+        Pull any pending messages from the message_generator, sending each
+        one to either the event bus, the command_async or both, depending on
+        the values of fire_all and command
```
Comment (twangboy) [inline salt/engines/slack_bolt_engine.py:824]: Docs for the args

File: salt/engines/slack_bolt_engine.py [hunk 11]
```diff
@@ -0,0 +1,1004 @@
...
+    control=False,
+    trigger="!",
+    groups=None,
+    groups_pillar_name=None,
+    fire_all=False,
+    tag="salt/engines/slack",
+):
+    """
+    Listen to slack events and forward them to salt, new version
+    """
```
Comment (twangboy) [inline salt/engines/slack_bolt_engine.py:988]: Docs for the args

**Review by twangboy — CHANGES_REQUESTED** [review]

**Review by twangboy — CHANGES_REQUESTED** [review]
Need to standardize on the type/parameter for all the arguments in the docs for the functions. You're doing it on some and not others.

**Review by Ch3LL — APPROVED** [review]

**Pull request merged**